docs/changes/units/17586.api.rst

```diff
@@ -0,0 +1,2 @@
+Automatic conversion of a ``str`` or ``bytes`` instance to a unit when it is
+multiplied or divided with an existing unit is deprecated.
```

Comment:
As far as I can tell this is an undocumented feature, so I don't think we need to justify the deprecation in the change log in great detail.

Reply:
But maybe do write "existing unit *or quantity*"